cirq/optimizers/stratify.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Type, Callable, Union, Iterable
+
+from cirq import ops, circuits
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def stratified_circuit(
```

Comment:
There is a well-known convention that calls for verb-based function names. I think we haven't consistently followed it, but here adherence is trivial: just rename to `stratify_circuit`.

This will make cirq feel more familiar to all folks who know the convention :-)

Reply:
The issue with `stratify_circuit` is that circuit is mutable, and that name implies we are mutating the original circuit instead of returning a new one.

Reply:
I guess `make_stratified_circuit` would work, but on second thought I think the current name fits well with python's other conventions (c.f. `sorted`, `reversed` etc).